### PR TITLE
Exposing autoComplete prop on SLDSInput and input prop on SLDSCombobox.

### LINKED ISSUE #1654: Expose input autocomplete attribute in Combobox and Input
Newer versions of Chrome completely ignore `autocomplete="off"`. So if you have something like a ComboBox of States and you want to allow a user to filter the options via input, Chrome might recognize the input as part of an Address and try to autofill the State.

A possible solution is to set `autocomplete="some_invalid_value"` which causes Chrome to not prompt for an autofill value.

The problem with that is, Firefox will default to using `autocomplete="on"` on invalid values.

Per https://developer.mozilla.org/en-US/docs/Web/Security/Securing_your_site/Turning_off_form_autocompletion:

> In some cases, the browser will continue suggesting autocompletion values even if the autocomplete  attribute is set to off. This unexpected behavior can be quite puzzling for developers. The trick to   really enforcing non-autocompletion is to assign an invalid value to the attribute, for example:
> 
> `autocomplete="nope"`
> 
> Since this value is not a valid one for the autocomplete attribute, the browser has no way to match it,  and stops trying to autocomplete the field. Please note that this does not work in Firefox; it completely disregards the invalid value and reverts to default autocompletion behavior.

In my mind, the only solution is to do a quick browser check before setting any hard values for `autocomplete` like https://github.com/salesforce/design-system-react/blob/master/components/combobox/combobox.jsx#L809.

Thanks

### PR DESCRIPTION
**Pull request by davidlygagnon**
Fixes #1654 

### Additional description

- Adding autoComplete prop on SLDSInput component. 
- Adding input prop on SLDSCombobox component. This will allow users to override the inner-input props of the combobox with the ones passed into the input component.
- Added one more test for the new combobox input prop.

---

### CONTRIBUTOR checklist (do not remove)
Please complete for every pull request

* [ ] First-time contributors should sign the Contributor License Agreement. It's a fancy way of saying that you are giving away your contribution to this project. If you haven't before, wait a few minutes and a bot will comment on this pull request with instructions.
* [ ] `npm run lint:fix` has been run and linting passes.
* [ ] Mocha, Jest (Storyshots), and `components/component-docs.json` CI tests pass (`npm test`).
* [ ] Tests have been added for new props to prevent regressions in the future. See [readme](https://github.com/salesforce/design-system-react/blob/master/tests/README.md).
* [ ] Review the appropriate Storybook stories. Open [http://localhost:9001/](http://localhost:9001/).
* [ ] The Accessibility panel of each Storybook story has 0 violations (aXe). Open [http://localhost:9001/](http://localhost:9001/).
* [ ] Review tests are passing in the browser. Open [http://localhost:8001/](http://localhost:8001/).
* [ ] Review markup conforms to [SLDS](https://www.lightningdesignsystem.com/) by looking at [DOM snapshot strings](https://facebook.github.io/jest/docs/en/snapshot-testing.html).

### REVIEWER checklist (do not remove)

* [ ] TravisCI tests pass. This includes linting, Mocha, Jest, Storyshots, and `components/component-docs.json` tests.
* [ ] Tests have been added for new props to prevent regressions in the future. See [readme](https://github.com/salesforce/design-system-react/blob/master/tests/README.md).
* [ ] Review the appropriate Storybook stories. Open [http://localhost:9001/](http://localhost:9001/).
* [ ] The Accessibility panel of each Storybook story has 0 violations (aXe). Open [http://localhost:9001/](http://localhost:9001/).
* [ ] Review tests are passing in the browser. Open [http://localhost:8001/](http://localhost:8001/).
* [ ] Review markup conforms to [SLDS](https://www.lightningdesignsystem.com/) by looking at [DOM snapshot strings](https://facebook.github.io/jest/docs/en/snapshot-testing.html).
###### Required only if there are markup / UX changes
* [ ] Add year-first date and commit SHA to `last-slds-markup-review` in `package.json` and push.
* [ ] Request a review of the deployed Heroku app by the Salesforce UX Accessibility Team.
* [ ] Add year-first review date, and commit SHA, `last-accessibility-review`, to `package.json` and push.
* [ ] While the contributor's branch is checked out, run `npm run local-update` within locally cloned [site repo](https://github.com/salesforce-ux/design-system-react-site) to confirm the site will function correctly at the next release.

### CODE REVIEW

File: components/input/index.jsx [hunk 1]
```diff
@@ -108,6 +108,10 @@ class Input extends React.Component {
 			label: PropTypes.string,
 			spinner: PropTypes.string,
 		}),
+		/**
+		 * Disabled brower's autocomplete when "off" is used.
+		 */
+		autoComplete: PropTypes.string,
```
Comment (davidlygagnon) [inline components/input/index.jsx]: Should I add a new test just for this prop ?

File: components/combobox/combobox.jsx [hunk 2]
```diff
@@ -139,6 +139,11 @@ const propTypes = {
 	 * HTML id for component. _Tested with snapshot testing._
 	 */
 	id: PropTypes.string,
+	/**
+	 * An [Input](https://react.lightningdesignsystem.com/components/inputs) component.
+	 * The props from this component will override any default props.
+	 */
+	input: PropTypes.node,
```
Comment (davidlygagnon) [inline components/combobox/combobox.jsx]: Now allowing to pass an input as a prop.

File: components/combobox/combobox.jsx [hunk 3]
```diff
@@ -1026,6 +1037,7 @@ class Combobox extends React.Component {
 									props.value
 								: props.value
 						}
+						{...userDefinedProps.input}
```
Comment (davidlygagnon) [inline components/combobox/combobox.jsx]: Input props override any props set above this line.

File: components/input/index.jsx (hunk 1 above)
Comment (interactivellama) [reply components/input/index.jsx]: All props that can be changed either need to be tracked via snapshot tests or mocha tests.

Comment (interactivellama) [reply components/input/index.jsx]: I think autocomplete='test' is in one of the snap shots.

Comment (interactivellama) [reply components/input/index.jsx]: Oh, I'm sorry. I misread this. I thought it was in combobox. Yes, a snapshot is fine.

**Pull request merged**